File: Practice Sorting/4_WA Test Cases.py
t = int(input())
for _ in range(t):
    n = int(input())
    s = list(map(int, input().split()))
    v = input()
    min_val=None
    # your code goes here
    for index,value in enumerate(v):
        if value=="0":
            if min_val is None:
                min_val=s[index]
            else:
                if min_val>s[index]:
                    min_val=s[index]
    print(min_val)

File: Practice Sorting/codigo_cs/4WA Test Cases.cs
using System;
using System.Linq;
public class Test4
{
    public static void WATestCases()
    {
        // your code goes here
        int t = int.Parse(Console.ReadLine());
        for (int i = 0; i < t; i++)
        {
            int n = int.Parse(Console.ReadLine());
            int[] s = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
            string v = Console.ReadLine();
            int minVal = 101;

            for (int j = 0; j < n; j++)
            {
                if (v[j] == '0' && s[j] < minVal)
                {
                    minVal = s[j];
                }
            }
            Console.WriteLine(minVal);

        }
    }
}
